src/core/server/saved_objects/service/lib/repository.ts

```diff
@@ -1005,8 +1005,13 @@ export class SavedObjectsRepository {
       throw SavedObjectsErrorHelpers.createGenericNotFoundError(type, id);
     }
 
-    const existingNamespaces = getResponse._source.namespaces as string[] | undefined;
-    if (existingNamespaces) {
+    const existingNamespaces = (getResponse?._source?.namespaces || []) as string[];
+    if (options.validateExistingNamespaces) {
+      await options.validateExistingNamespaces(existingNamespaces);
```

Comment:
Note: the way this is designed, `validateExistingNamespaces` will get called with an empty array if the saved object does not have any existing namespaces. As mentioned in the comment below, there should never be a case where this happens in normal operations, but it is a possibility if someone manually modifies the document in Elasticsearch.

We deliberately check permissions even if there are no existing namespaces, and `CheckSavedObjectsPrivileges` fails safe when attempting to check permissions in 0 namespaces. In other words: if the multi-namespace saved object doesn't exist in any namespace, nobody is authorized to update it.